src/main/java/com/_up/megastore/utils/EmailBuilder.java

```diff
@@ -114,19 +123,18 @@ public String buildNewActivationEmail(User user, UUID activationToken) {
                 + "</table>";
     }
 
-    public String buildFinishOrderEmail(Order order) {
+    public String buildOrderEmail(Order order, String subject) {
         String orderDetailUrl = frontendURL + "/orders/" + order.getOrderId();
 
         return "<table style='width:100%; height:100%;'>"
                 + "<tr><td style='width:100%; height:100%; text-align:center; vertical-align:middle;'>"
                 + "<div style='display:inline-block;'>"
-                + "<h1>Order Finished</h1>"
+                + "<h1>" + subject + "</h1>"
                 + "<h3>Hey " + order.getUser().getFullName() + "!</h3>"
-                + "<h4>Your order of the day " + order.getDate() + " has been finished.</h4>"
+                + "<h4>Your order of the day " + order.getDate() + STATE_MESSAGES.get(order.getState()) + "</h4>"
```

Comment:
Y esto lo reemplazariamos por

```suggestion
                + "<h4>Your order of the day " + order.getDate() + order.getState().stateMessage + "</h4>"
```